source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -51,34 +51,19 @@ void GrpcClientImpl::onSuccess(std::unique_ptr<envoy::service::auth::v3::CheckRe
     span.setTag(TracingConstants::get().TraceStatus, TracingConstants::get().TraceOk);
     authz_response->status = CheckStatus::OK;
     if (response->has_ok_response()) {
-      toAuthzResponseHeader(authz_response, response->ok_response().headers());
-      if (response->ok_response().headers_to_remove_size() > 0) {
-        for (const auto& header : response->ok_response().headers_to_remove()) {
-          authz_response->headers_to_remove.push_back(Http::LowerCaseString(header));
-        }
+      const auto& ok_response = response->ok_response();
+      copyHeaderMutationsIntoResponse(authz_response, ok_response.headers(),
+                                      ok_response.response_headers_to_add(),
+                                      ok_response.headers_to_remove());
+
+      for (const auto& query_parameter : ok_response.query_parameters_to_set()) {
+        authz_response->query_parameters_to_set.emplace_back(
+            Http::Utility::PercentEncoding::urlEncodeQueryParameter(query_parameter.key()),
```

Comment:
Should we use runtime guard for the change of using URL-encoding? This will introduces the behavior change.

Reference in the past: `envoy_reloadable_features_oauth_use_url_encoding`

Reply:
Done.